src/common/src/mlib/duration.h

```diff
@@ -0,0 +1,386 @@
+static inline mlib_duration
+_mlibDurationMultiply (const mlib_duration dur, int fac) mlib_noexcept
+{
+   mlib_duration ret = {0};
+   if (mlib_mul (&ret._rep, dur._rep, fac)) {
+      if ((dur._rep < 0) != (fac < 0)) {
+         // Different signs:  Neg × Pos = Neg
+         ret = mlib_duration_min ();
+      } else {
+         // Same signs: Pos × Pos = Pos
+         //             Neg × Neg = Pos
+         ret = mlib_duration_max ();
+      }
+   }
+   return ret;
+}
+
+// Duration scalar divide
+#define _mlibDurationInfixOperator_div(LHS, Div) _mlibDurationDivide (_mlibDurationArgument (LHS), (Div))
+static inline mlib_duration
+_mlibDurationDivide (mlib_duration a, int div) mlib_noexcept
+{
+   mlib_check (div, neq, 0);
+   if (div == -1 && a._rep == mlib_minof (mlib_duration_rep_t)) {
+      // MIN / -1 is UB, but the saturating result is the max
+      a = mlib_duration_max ();
+   } else {
+      a._rep /= div;
+   }
+   return a;
+}
+
+// Duration addition
+#define _mlibDurationInfixOperator_plus(LHS, RHS) \
+   _mlibDurationAdd (_mlibDurationArgument (LHS), _mlibDurationArgument (RHS))
+static inline mlib_duration
+_mlibDurationAdd (const mlib_duration a, const mlib_duration b) mlib_noexcept
+{
+   mlib_duration ret = {0};
+   if (mlib_add (&ret._rep, a._rep, b._rep)) {
+      if (a._rep > 0) {
+         ret = mlib_duration_max ();
+      } else {
+         ret = mlib_duration_min ();
+      }
+   }
+   return ret;
+}
+
+// Duration subtraction
+#define _mlibDurationInfixOperator_minus(LHS, RHS) \
+   _mlibDurationSubtract (_mlibDurationArgument (LHS), _mlibDurationArgument (RHS))
+static inline mlib_duration
+_mlibDurationSubtract (const mlib_duration a, const mlib_duration b) mlib_noexcept
+{
+   mlib_duration ret = {0};
+   if (mlib_sub (&ret._rep, a._rep, b._rep)) {
+      if (a._rep < 0) {
+         ret = mlib_duration_min ();
+      } else {
+         ret = mlib_duration_max ();
+      }
+   }
+   return ret;
+}
+
+#define _mlibDurationInfixOperator_min(Duration, RHS) \
+   _mlibDurationMinBetween (_mlibDurationArgument (Duration), _mlibDurationArgument (RHS))
+static inline mlib_duration
+_mlibDurationMinBetween (mlib_duration lhs, mlib_duration rhs)
+{
+   if (lhs._rep < rhs._rep) {
+      return lhs;
+   }
+   return rhs;
+}
+
+#define _mlibDurationInfixOperator_max(Duration, RHS) \
+   _mlibDurationMaxBetween (_mlibDurationArgument (Duration), _mlibDurationArgument (RHS))
+static inline mlib_duration
+_mlibDurationMaxBetween (mlib_duration lhs, mlib_duration rhs)
+{
+   if (lhs._rep > rhs._rep) {
+      return lhs;
+   }
+   return rhs;
+}
+
+/**
+ * @brief Create a duration object from a count of some unit of time
+ *
+ * @param Count An integral expression
+ * @param Unit A unit suffix identifier, must be one of:
+ *
+ * - `ns` (nanoseconds)
+ * - `us` (microseconds)
+ * - `ms` (milliseconds)
+ * - `sec` (seconds)
+ * - `min` (minutes)
```

Comment:
```suggestion
 * - `ns` (nanoseconds)
 * - `us` (microseconds)
 * - `ms` (milliseconds)
 * - `s` (seconds)
 * - `min` (minutes)
 * - `h` (hours)
```
Consider `sec` -> `s` and adding `h` (hours) to match the set of `std::chrono::duration` [UDL suffixes](https://eel.is/c++draft/time.duration.literals).